### PR TITLE
Notification Comments: add wrapper view controller to display details

### PR DESCRIPTION
**Pull request by ScoutHarris**
Ref: #17790 

To facilitate showing a `NoResultsView` when loading Comments for Notifications, a new `NotificationCommentDetailViewController` has been added. It is used to load/fetch the Comment for display, and show the `CommentDetailViewController` when the Comment has been obtained. 

- The logic for loading/fetching Comments and displaying `CommentDetailViewController` has been moved from `NotificationCommentDetailCoordinator` into the new VC.
- Notification navigation button logic has been moved from `CommentDetailViewController` into the new VC.
- Showing the `NoResultsView` with the notification navigation buttons proved persnickity, so I'll do that in a follow-up.
  - Because there is no loading view, sometimes there is a delay and/or a blank view is displayed after selecting a notification while the Comment is loaded.
- Sometimes when a Comment is fetched, there's a gap at the top of the details view. Once the Comment is cached, and the comment is redisplayed, the layout is fine. Pretty sure I've seen it before, so I think it's legacy. I'll look at it separately. 

To test:
TLDR: everything should work as it did before. 😄 

- Go to Notifications.
- Select a Comment notification.
- Verify the Edit, Previous, Next nav bar buttons work as expected.
- Verify the Previous/Next buttons are disabled when viewing the first/last notification.
- Trash/Spam a comment. Verify the Notification is removed from the list.
- Delete a comment. Verify the Notification is removed from the list.
- Verify My Site Comments look & behave correctly.

## Regression Notes
1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Notifications/Tools/NotificationCommentDetailCoordinator.swift [hunk 1]
```diff
@@ -166,21 +72,17 @@ private extension NotificationCommentDetailCoordinator {
...
 
         guard let viewController = viewController,
-        let notificationDetailsViewController = storyboard.instantiateViewController(withIdentifier: Notifications.viewControllerName) as? NotificationDetailsViewController else {
-            DDLogError("NotificationCommentDetailCoordinator: missing view controller.")
-            return
-        }
+              let notificationDetailsViewController = storyboard.instantiateViewController(withIdentifier: Notifications.viewControllerName) as? NotificationDetailsViewController else {
+                  DDLogError("NotificationCommentDetailCoordinator: missing view controller.")
+                  return
+              }
```
Comment (dvdchr) [inline WordPress/Classes/ViewRelated/Notifications/Tools/NotificationCommentDetailCoordinator.swift:85]: super minor indentation thingy:
```suggestion
        let notificationDetailsViewController = storyboard.instantiateViewController(withIdentifier: Notifications.viewControllerName) as? NotificationDetailsViewController else {
            DDLogError("NotificationCommentDetailCoordinator: missing view controller.")
            return
        }
```

File: WordPress/Classes/ViewRelated/Notifications/Tools/NotificationCommentDetailCoordinator.swift [hunk 2]
```diff
@@ -274,9 +162,9 @@ extension NotificationCommentDetailCoordinator: CommentDetailsNotificationDelega
 
     func commentWasModerated(for notification: Notification?) {
         guard let notification = notification,
-        !notificationsCommentModerated.contains(notification) else {
-            return
-        }
+              !notificationsCommentModerated.contains(notification) else {
+                  return
+              }
```
Comment (dvdchr) [inline WordPress/Classes/ViewRelated/Notifications/Tools/NotificationCommentDetailCoordinator.swift:167]:
```suggestion
        !notificationsCommentModerated.contains(notification) else {
            return
        }
```

**Review by dvdchr — APPROVED** [review]
Awesome! I've tested the capabilities in both Notification and My Sites, and both work well. I've found a couple of visual issues, but I'm approving to unblock + this being a WIP.

When loading a comment from remote, I noticed a minor oddity during view transition, which is most likely due to transparent background color:

Simulator Screen Shot - iPhone 8 - 2022-03-01 at 23 12 28

I assume this is expected because the loading (and/or error) view is not hooked yet, but leaving a comment here just in case. 🙂 

Additionally, I've also found an issue where navigating to a Notification Comment Detail from a non-Comment detail via pagination causes the Comment Detail view to have a gap on top:

Simulator Screen Shot - iPhone 8 - 2022-03-01 at 23 23 11

Repro steps:
1. Select a non-comment notification.
2. Navigate to a comment notification with pagination. The comment needs to be fetched from remote.
3. Observe that there is a gap on top of the comment detail.

File: WordPress/Classes/ViewRelated/Notifications/Tools/NotificationCommentDetailCoordinator.swift (hunk 1 above)
Comment (ScoutHarris) [reply WordPress/Classes/ViewRelated/Notifications/Tools/NotificationCommentDetailCoordinator.swift:85]: Hrm. Xcode indents it my way, not your way. I'm leaving it. I'm not going back and forth because Xcode can't make up it's mind. 😆

File: WordPress/Classes/ViewRelated/Notifications/Tools/NotificationCommentDetailCoordinator.swift (hunk 2 above)
Comment (ScoutHarris) [reply WordPress/Classes/ViewRelated/Notifications/Tools/NotificationCommentDetailCoordinator.swift:167]: Ditto.

**Comment by ScoutHarris** [comment]
> When loading a comment from remote, I noticed a minor oddity during view transition
> I assume this is expected because the loading (and/or error) view is not hooked yet

Yea, that's what I meant by `sometimes there is a delay and/or a blank view is displayed`. That will go away when I add the loading view.

> navigating to a Notification Comment Detail from a non-Comment detail via pagination causes the Comment Detail view to have a gap on top

That's related to my note about `Sometimes when a Comment is fetched, there's a gap at the top of the details view`. On my list to look at.

Thanks @dvdchr !

**Comment by dvdchr** [comment]
> Yea, that's what I meant by sometimes there is a delay and/or a blank view is displayed. That will go away when I add the loading view.
> ...
> navigating to a Notification Comment Detail from a non-Comment detail via pagination causes the Comment Detail view to have a gap on top

Whoops, sorry for missing those remarks! I swear I've read the description, but my brain decided to play tricks on me. 😅
